core/src/core/pmmr/segment.rs

```diff
@@ -69,6 +69,28 @@ impl Writeable for SegmentIdentifier {
 	}
 }
 
+impl SegmentIdentifier {
+	/// Test helper to get an iterator of SegmentIdentifiers required to read a
+	/// pmmr of size `target_mmr_size` in segments of height `segment_height`
+	pub fn traversal_iter(
+		target_mmr_size: u64,
+		segment_height: u8,
+	) -> impl Iterator<Item = SegmentIdentifier> {
+		(0..SegmentIdentifier::count_segments_required(target_mmr_size, segment_height)).map(
+			move |idx| SegmentIdentifier {
+				height: segment_height,
+				idx: idx as u64,
+			},
+		)
+	}
+
+	/// Returns number of segments required that would needed in order to read a
+	/// pmmr of size `target_mmr_size` in segments of height `segment_height`
+	pub fn count_segments_required(target_mmr_size: u64, segment_height: u8) -> usize {
+		pmmr::n_leaves(target_mmr_size) as usize / (1 << segment_height as usize) + 1
```

Comment:
why + 1 ?
Do you add an empty segment if mmr size is an exact multiple of 2^segment_height ?

Reply:
Left over from earlier hack-test, thanks for spotting